src/analysis/retail/evoker/devastation/modules/guide/EssenceGraph/EssenceGraph.tsx

```diff
@@ -0,0 +1,65 @@
+import EssenceTracker from 'analysis/retail/evoker/preservation/modules/features/EssenceTracker';
+import ResourceGraph from 'parser/shared/modules/ResourceGraph';
+import { formatTime } from 'parser/ui/BaseChart';
+
+class EssenceGraph extends ResourceGraph {
+  static dependencies = {
+    ...ResourceGraph.dependencies,
+    essenceTracker: EssenceTracker,
+  };
+
+  get vegaSpec() {
+    return {
+      data: {
+        name: 'graphData',
+      },
+      transform: [
+        {
+          filter: 'isValid(datum.timestamp)',
+        },
+        {
+          calculate: `datum.timestamp - ${this.owner.fight.start_time}`,
+          as: 'timestamp_shifted',
+        },
+      ],
+      encoding: {
+        x: {
+          field: 'timestamp_shifted',
+          type: 'quantitative' as const,
+          axis: {
+            labelExpr: formatTime('datum.value'),
+            tickCount: 25,
+            grid: false,
+          },
+          scale: {
+            nice: false,
+          },
+          title: null,
+        },
+        y: {
+          field: 'amount',
+          type: 'quantitative' as const,
+        },
+      },
+      resolve: {
+        scale: { y: 'independent' as const },
+      },
+      mark: {
+        type: 'line' as const,
+        interpolate: 'step' as const,
+        color: '#33937F' as const,
+      },
+      config: {
+        view: {},
+      },
+    };
+  }
```

Comment:
What's the custom `vegaSpec` for? If it's just for custom color you can use a `lineColor` override, see RunicPowerGraph for an example.